web/src/featureToggle.ts

```diff
@@ -1,19 +1,29 @@
 /* eslint no-console: 0 */
-/* eslint no-throw-literal: 0 */
 import Vue from 'vue';
+import { reactive } from '@vue/composition-api';
+
+// Declare additions to the global `window` object
+declare global {
+  interface Window {
+    enable: (toggle: string) => void,
+    disable: (toggle: string) => void,
+  }
+}
 
 // Create a component to hold the feature toggles reactively
 // This object can be mutated at runtime to enable/disable feature toggles
-const featureToggles = new Vue({
-  data() {
-    return {
-      DJANGO_API: false,
-    };
-  },
+const featureToggles = reactive({
+  DJANGO_API: false,
 });
 
+// Used to narrow string types
+function isToggle(toggle: string): toggle is keyof (typeof featureToggles) {
+  return Object.keys(featureToggles).includes(toggle);
+}
+
 // Register the feature toggles as computed properties on all components
 // Now feature toggles are available on all components and templates without any imports
+// TODO: Convert this to use composition API.
```

Comment:
Probably should file an issue too.

Reply:
I'm not sure if this is possible. Composition API is an alternative way to write components, but this really doesn't have anything to do with that. This injects a computed variable called `DJANGO_API` onto every component so the DJANGO_API toggle can be used in templates. I don't think Composition API introduces an alternative way to do this, even if this does use the dreaded `mixin` keyword.

Reply:
The normal way to solve this is by declaring a composition-api computed variable here named `DJANGO_API`, and importing it wherever it's needed. While this involves an additional import everywhere it's used, that actually doesn't bother me. The problem is that any components which want to import this computed variable must use composition API, or use the normal options `computed` block to wrap it. I didn't think this was worth the effort since it creeps the scope somewhat, and the toggle will likely be removed soon anyway. However if you think it's worth converting this as well, I'd be happy to.